File: rag-evaluation-backend/app/schemas/dataset.py
from typing import Optional, List, Dict, Any
from datetime import datetime
from pydantic import BaseModel, Field, ConfigDict
from uuid import UUID

# 基础数据集模型
class DatasetBase(BaseModel):
    name: str
    description: Optional[str] = None
    is_public: Optional[bool] = False
    tags: Optional[List[str]] = []
    dataset_metadata: Optional[Dict[str, Any]] = {}

# 创建数据集时的请求模型
class DatasetCreate(DatasetBase):
    pass

# 更新数据集时的请求模型
class DatasetUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    is_public: Optional[bool] = None
    tags: Optional[List[str]] = None
    dataset_metadata: Optional[Dict[str, Any]] = None

# 数据集响应模型
class DatasetOut(DatasetBase):
    id: str
    user_id: Optional[str] = None
    question_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

# 数据集详情响应模型
class DatasetDetail(DatasetOut):
    projects: List[Dict[str, str]] = []
    
    model_config = ConfigDict(from_attributes=True)

# 项目-数据集关联模型
class ProjectDatasetLink(BaseModel):
    project_id: str
    dataset_id: str

    model_config = ConfigDict(from_attributes=True)

# 批量关联数据集请求
class BatchLinkDatasets(BaseModel):
    dataset_ids: List[str] 

File: rag-evaluation-frontend/src/types/dataset.ts
export interface Dataset {
  id: string;
  user_id: string;
  name: string;
  description?: string;
  is_public: boolean;
  tags: string[];
  dataset_metadata: Record<string, any>;
  question_count: number;
  created_at: string;
  updated_at: string;
}

export interface DatasetDetail extends Dataset {
  projects: Array<{
    id: string;
    name: string;
  }>;
}

export interface Question {
  id: string;
  dataset_id: string;
  question_text: string;
  standard_answer: string;
  category?: string;
  difficulty?: string;
  type?: string;
  tags?: Record<string, any>;
  question_metadata?: Record<string, any>;
  created_at: string;
  updated_at: string;
}

export interface CreateDatasetRequest {
  name: string;
  description?: string;
  is_public: boolean;
  tags?: string[];
  dataset_metadata?: Record<string, any>;
}

export interface UpdateDatasetRequest {
  name?: string;
  description?: string;
  is_public?: boolean;
  tags?: string[];
  dataset_metadata?: Record<string, any>;
}

export interface DatasetListParams {
  page?: number;
  size?: number;
  search?: string;
  tag?: string;
  is_public?: boolean;
}

export interface ImportDataRequest {
  dataset_id: string;
  file?: File;
  questions?: Array<{
    question_text: string;
    standard_answer: string;
    category?: string;
    difficulty?: string;
    tags?: Record<string, any>;
  }>;
}

export interface ProjectDatasetLink {
  project_id: string;
  dataset_id: string;
}

export interface BatchLinkDatasetsRequest {
  project_id: string;
  dataset_ids: string[];
}

export interface RagAnswer {
  id: string;
  question_id: string;
  answer_text: string;
  source_type: string;
  version: string;
  api_config_id?: string;
  response_time?: number;
  token_count?: number;
  metadata?: any;
  created_at: string;
  updated_at: string;
}

export interface CreateRagAnswerRequest {
  question_id: string;
  answer_text: string;
  source_type: string;
  version?: string;
  api_config_id?: string;
  response_time?: number;
  token_count?: number;
  metadata?: any;
}

export interface ImportQuestionWithRagRequest {
  dataset_id: string;
  questions: {
    question_text: string;
    standard_answer: string;
    category?: string;
    difficulty?: string;
    type?: string;
    tags?: string[];
    rag_answer?: CreateRagAnswerRequest;
  }[];
} 